src/Raven.Client/Util/AtomicDictionary.cs

```diff
@@ -7,10 +7,10 @@
 
 namespace Raven.Client.Util
 {
-    internal class AtomicDictionary<TVal> : IEnumerable<KeyValuePair<string, TVal>>
+    internal class AtomicDictionary<TVal> : IEnumerable<KeyValuePair<RequestDestination, TVal>>
```

Comment:
should be TKey, but I do not think we need `AtomicDictionary` at this point? Are we using any of it's features? This can be just a concurrent dictionary? cc @ayende

Reply:
done